src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -188,17 +188,18 @@ int sd_dhcp6_client_set_mac(
 
         assert_return(client, -EINVAL);
         assert_return(addr, -EINVAL);
-        assert_return(addr_len > 0 && addr_len <= MAX_MAC_ADDR_LEN, -EINVAL);
-        assert_return(arp_type > 0, -EINVAL);
 
         assert_return(IN_SET(client->state, DHCP6_STATE_STOPPED), -EBUSY);
 
         if (arp_type == ARPHRD_ETHER)
```

Comment:
For tun devices it is `ARPHRD_EETHER`, which seems bogus. The value comes from `iftype`